### PR TITLE
Initial working implementation of vms

### PR DESCRIPTION
**Pull request by thedevelopnik**
Some key files are left out to maintain secrets until I get some scripts that add the vars in from the command line. And the providers/backend file has secrets in it as well, so left that out for now.

The magic here is in the cloud_init_config object which lets me conveniently put startup scripts in. tbot and teleport start up automagically now with no manual work required.

### CODE REVIEW

File: terraform/vms.tf [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    description      = "SSH from anywhere"
+    from_port        = 22
+    to_port          = 22
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
```
Comment (benarent) [inline terraform/vms.tf]: restrict this to a home IP, if wanting to not post on the internet please put in a secrets file

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+prereqs
+.terraform.lock.hcl
+.terraform
+vars.tf
+main.tf
```
Comment (benarent) [inline .gitignore]:
```suggestion
main.tf
errored.tfstate
```

File: backend1/main.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	defer source.Close()
+
+	svid, err := source.GetX509SVID()
+	if err != nil {
+		return fmt.Errorf("unable to get X509SVID: %w", err)
+	}
+	fmt.Println(svid.ID.String())
+
+	clientID := spiffeid.RequireFromString("spiffe://teleport-ent-15-workload.asteroid.earth/client")
```
Comment (benarent) [inline backend1/main.go]: Can you make this a variable so it's easier for other people to demo/switch out.

File: testing/create-backend-bot.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#! /usr/bin/env bash
+
+set -euxo pipefail
+
+mkdir /etc/tbot
+TOKEN=$(tctl bots add workload-id-demo-backend-2-bot --roles=workload-id-demo-backend-1-spiffe-bot --format=json | jq -r '.token_id')
+WORKLOAD_IDENTITY_EXPERIMENT=1 tbot start \
+   --destination-dir=/opt/machine-id/demo-backend-1-bot \
+   --token="$TOKEN" \
+   --auth-server=teleport-ent-15.asteroid.earth:443 \
```
Comment (benarent) [inline testing/create-backend-bot.sh:11]: Another variable cleanup task to make this easier to share / use in the future.

File: .gitignore (hunk 2 above)
Comment (thedevelopnik) [reply .gitignore]: Done!

File: terraform/vms.tf (hunk 1 above)
Comment (thedevelopnik) [reply terraform/vms.tf]: Removed it entirely now that I have Teleport working and I can get in that way.

File: backend1/main.go (hunk 3 above)
Comment (thedevelopnik) [reply backend1/main.go]: Done, made a lot of stuff a variable.

**Pull request merged**